src/Microsoft.DotNet.Watcher.Core/Internal/Implementation/Project.cs

```diff
@@ -15,12 +17,27 @@ public Project(ProjectModel.Project runtimeProject)
             ProjectFile = runtimeProject.ProjectFilePath;
             ProjectDirectory = runtimeProject.ProjectDirectory;
 
-            Files = runtimeProject.Files.SourceFiles.Concat(
-                    runtimeProject.Files.ResourceFiles.Values.Concat(
-                    runtimeProject.Files.PreprocessSourceFiles.Concat(
-                    runtimeProject.Files.SharedFiles))).Concat(
-                    new string[] { runtimeProject.ProjectFilePath })
-                .ToList();
+            var compilerOptions = runtimeProject.GetCompilerOptions(targetFramework: null, configurationName: null);
+
+            var filesToWatch = new List<string>() { runtimeProject.ProjectFilePath };
+            if (compilerOptions.CompileInclude != null)
```

Comment:
I don't know about the object model here, but is `CompileInclude` the final list of all files that need to be compiled? (i.e considering Include, IncludeFiles,Exclude,ExcludeFiles etc.)